### PR TITLE
Math.sign and Math.trunc

### PR DESCRIPTION
**Pull request by orkvi**
JerryScript-DCO-1.0-Signed-off-by: Virag Orkenyi orkvi@inf.u-szeged.hu

### CODE REVIEW

File: jerry-core/ecma/builtin-objects/ecma-builtin-math.c [hunk 1]
```diff
@@ -152,7 +154,69 @@ ecma_builtin_math_object_max_min (bool is_max, /**< 'max' or 'min' operation */
...
+/**
+ * The Number.prototype object's 'Math.trunc' routine
+ *
+ * See also:
+ *          ECMA-262 v6, 20.2.2.35
+ *
+ * @return ecma value
+ *         Returned value must be freed with ecma_free_value.
+ */
+static ecma_number_t
```
Comment (rerobika) [inline jerry-core/ecma/builtin-objects/ecma-builtin-math.c]: These all new functions/switch cases must be guarded with `JERRY_ES2015_BUILTNS` macro.

File: jerry-core/ecma/builtin-objects/ecma-builtin-math.c [hunk 2]
```diff
@@ -55,10 +55,12 @@ enum
   ECMA_MATH_OBJECT_EXP, /* ECMA-262 v5, 15.8.2.8 */
   ECMA_MATH_OBJECT_FLOOR, /* ECMA-262 v5, 15.8.2.9 */
   ECMA_MATH_OBJECT_LOG, /* ECMA-262 v5, 15.8.2.10 */
+  ECMA_MATH_OBJECT_TRUNC, /* ECMA-262 v5, 20.2.2.35  */
```
Comment (rerobika) [inline jerry-core/ecma/builtin-objects/ecma-builtin-math.c]: v5->v6

File: jerry-core/ecma/builtin-objects/ecma-builtin-math.c [hunk 3]
```diff
@@ -153,6 +155,79 @@ ecma_builtin_math_object_max_min (bool is_max, /**< 'max' or 'min' operation */
...
+ecma_builtin_math_object_sign (ecma_number_t arg)
+{
+  if (ecma_number_is_nan (arg))
+  {
+    return arg;
+  }
+
+  if (ecma_number_is_negative (arg))
+  {
+    if (ecma_number_is_zero (arg))
```
Comment (dbatyai) [inline jerry-core/ecma/builtin-objects/ecma-builtin-math.c]: Would it be possible to merge the positive and negative `is_zero` checks and return with `arg`? Or even merge it into the `is_nan` check.

File: jerry-core/ecma/builtin-objects/ecma-builtin-math.c [hunk 4]
```diff
@@ -153,6 +155,79 @@ ecma_builtin_math_object_max_min (bool is_max, /**< 'max' or 'min' operation */
...
+ * See also:
+ *          ECMA-262 v6, 20.2.2.35
+ *
+ * @return ecma value
+ *         Returned value must be freed with ecma_free_value.
+ */
+static ecma_number_t
+ecma_builtin_math_object_trunc (ecma_number_t arg)
+{
+  if (ecma_number_is_nan (arg)||ecma_number_is_infinity (arg)||ecma_number_is_zero (arg))
```
Comment (dbatyai) [inline jerry-core/ecma/builtin-objects/ecma-builtin-math.c]: Please add whitespaces around the `||` operators.

File: tests/jerry/es2015/math-sign.js [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+// distributed under the License is distributed on an "AS IS" BASIS
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+var nan = NaN;
+var p_zero = 0.0;
+var m_zero = -p_zero;
+
+
```
Comment (dbatyai) [inline tests/jerry/es2015/math-sign.js]: Please remove this extra newline.

File: tests/jerry/es2015/math-trunc.js [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+assert (Math['trunc'](m_inf) === m_inf);
+
+assert (Math['trunc'](0.5) === p_zero);
+assert (Math['trunc'](-0.5) === m_zero);
+assert (Math['trunc'](1.2) === 1);
+assert (Math['trunc'](-1.5) === -1);
+assert (Math['trunc'](65.7) === 65);
+assert (Math['trunc'](-24.9) === -24);
+
+
```
Comment (dbatyai) [inline tests/jerry/es2015/math-trunc.js]: Please remove the trailing newlines.

File: jerry-core/ecma/builtin-objects/ecma-builtin-math.c [hunk 7]
```diff
@@ -55,11 +55,16 @@ enum
...
+  #if ENABLED (JERRY_ES2015_BUILTIN)
+  ECMA_MATH_OBJECT_TRUNC, /* ECMA-262 v6, 20.2.2.35  */
+  #endif /* ENABLED (JERRY_ES2015_BUILTIN) */
   ECMA_MATH_OBJECT_ROUND, /* ECMA-262 v5, 15.8.2.15 */
   ECMA_MATH_OBJECT_SIN, /* ECMA-262 v5, 15.8.2.16 */
   ECMA_MATH_OBJECT_SQRT, /* ECMA-262 v5, 15.8.2.17 */
   ECMA_MATH_OBJECT_TAN, /* ECMA-262 v5, 15.8.2.18 */
-
+  #if ENABLED (JERRY_ES2015_BUILTIN)
+  ECMA_MATH_OBJECT_SIGN, /* ECMA-262 v6, 20.2.2.29 */
```
Comment (rerobika) [inline jerry-core/ecma/builtin-objects/ecma-builtin-math.c]: Is there any reason why the 2 new methods not placed next to each other?

File: jerry-core/ecma/builtin-objects/ecma-builtin-math.c [hunk 8]
```diff
@@ -153,6 +158,65 @@ ecma_builtin_math_object_max_min (bool is_max, /**< 'max' or 'min' operation */
   return ecma_make_number_value (result_num);
 } /* ecma_builtin_math_object_max_min */
 
+#if ENABLED (JERRY_ES2015_BUILTIN)
+
+/**
+ * The Number.prototype object's 'Math.trunc' routine
```
Comment (rerobika) [inline jerry-core/ecma/builtin-objects/ecma-builtin-math.c]: The Math object's 'trunc' routine

File: jerry-core/ecma/builtin-objects/ecma-builtin-math.c [hunk 9]
```diff
@@ -153,6 +158,65 @@ ecma_builtin_math_object_max_min (bool is_max, /**< 'max' or 'min' operation */
...
+  if ((arg < 0) && (arg > -1))
+  {
+    return (ecma_number_t) -0;
+  }
+
+  return (ecma_number_t) arg - fmod (arg,1);
+} /* ecma_builtin_math_object_trunc */
+
+/**
+ * The Number.prototype object's 'Math.sign' routine
```
Comment (rerobika) [inline jerry-core/ecma/builtin-objects/ecma-builtin-math.c]: Ditto.

File: jerry-core/ecma/builtin-objects/ecma-builtin-math.c [hunk 10]
```diff
@@ -153,6 +158,65 @@ ecma_builtin_math_object_max_min (bool is_max, /**< 'max' or 'min' operation */
...
+  {
+    return (ecma_number_t) 0;
+  }
+
+  if ((arg < 0) && (arg > -1))
+  {
+    return (ecma_number_t) -0;
+  }
+
+  return (ecma_number_t) arg - fmod (arg,1);
```
Comment (rerobika) [inline jerry-core/ecma/builtin-objects/ecma-builtin-math.c]: missing space after `arg`

File: jerry-core/ecma/builtin-objects/ecma-builtin-math.c [hunk 11]
```diff
@@ -282,6 +346,17 @@ ecma_builtin_math_dispatch_routine (uint16_t builtin_routine_id, /**< built-in w
         x = DOUBLE_TO_ECMA_NUMBER_T (log (x));
         break;
       }
+
+      #if ENABLED (JERRY_ES2015_BUILTIN)
```
Comment (rerobika) [inline jerry-core/ecma/builtin-objects/ecma-builtin-math.c]: preprocessor directives must be in the first column. (same as below)

File: jerry-core/ecma/builtin-objects/ecma-builtin-math.c [hunk 12]
```diff
@@ -282,6 +344,22 @@ ecma_builtin_math_dispatch_routine (uint16_t builtin_routine_id, /**< built-in w
         x = DOUBLE_TO_ECMA_NUMBER_T (log (x));
         break;
       }
+
+#if ENABLED (JERRY_ES2015_BUILTIN)
```
Comment (dbatyai) [inline jerry-core/ecma/builtin-objects/ecma-builtin-math.c]: Please remove the empty lines around the preprocessor directives, here and below.

File: tests/jerry/es2015/math-trunc.js [hunk 13]
```diff
@@ -0,0 +1,32 @@
...
+assert (Math['trunc'](m_zero) === m_zero);
+assert (Math['trunc'](p_inf) === p_inf);
+assert (Math['trunc'](m_inf) === m_inf);
+
+assert (Math['trunc'](0.5) === p_zero);
+assert (Math['trunc'](-0.5) === m_zero);
+assert (Math['trunc'](1.2) === 1);
+assert (Math['trunc'](-1.5) === -1);
+assert (Math['trunc'](65.7) === 65);
+assert (Math['trunc'](-24.9) === -24);
```
Comment (dbatyai) [inline tests/jerry/es2015/math-trunc.js]: Files should contain exactly one newline at the end, please fix it.

Comment (rerobika) [reply tests/jerry/es2015/math-trunc.js]: @orkvi This request has not been fixed yet.

File: jerry-core/ecma/builtin-objects/ecma-builtin-math.c [hunk 14]
```diff
@@ -55,6 +55,10 @@ enum
   ECMA_MATH_OBJECT_EXP, /* ECMA-262 v5, 15.8.2.8 */
   ECMA_MATH_OBJECT_FLOOR, /* ECMA-262 v5, 15.8.2.9 */
   ECMA_MATH_OBJECT_LOG, /* ECMA-262 v5, 15.8.2.10 */
+  #if ENABLED (JERRY_ES2015_BUILTIN)
```
Comment (rerobika) [inline jerry-core/ecma/builtin-objects/ecma-builtin-math.c]: The macro guards are still misaligned.

**Pull request merged**